src/utils/PayloadValidation.ts

```diff
@@ -0,0 +1,28 @@
+import * as Joi from "@hapi/joi";
+import {adrValidation} from "./AdrValidation";
+
+const techRecordValidation = Joi.object().keys({
+  reasonForCreation: Joi.string().max(60).required(),
+  adrDetails: adrValidation
+});
+
+export const validatePayload = (payload: any) => {
+  let isTankOrBattery = false;
+  let isBattery = false;
+  if (payload.adrDetails && payload.adrDetails.vehicleDetails && payload.adrDetails.vehicleDetails.type) {
+    const vehicleDetailsType = payload.adrDetails.vehicleDetails.type.toLowerCase();
+    if (vehicleDetailsType.indexOf("battery") !== -1) {
```

Comment:
Technically missing a test case. 
![image](https://user-images.githubusercontent.com/50110006/70147295-d6efc080-169b-11ea-9fac-bc4f7c5fa1ef.png)
Somewhat trivial, but suggests the case of isBattery =  true isn't being tested elsewhere

Reply:
fixed